src/workerd/io/io-context.c++

```diff
@@ -292,6 +292,7 @@ IoContext::IncomingRequest::~IoContext_IncomingRequest() noexcept(false) {
 IoContext::~IoContext() noexcept(false) {
   // Kill the sentinel so that no weak references can refer to this IoContext anymore.
   selfRef->kill();
+  traceAsyncContextKey->reset();
```

Comment:
Not sure this actually helps, if the key is already in maps.

I think @jclee is planning to move this to be a singleton for the isolate rather than per-IoContext, which seems like the right fix.

Reply:
Ah, I think the idea is that if the key is already in maps, that by marking it dead with `reset()` it will eventually get filtered out by `AsyncContextFrame::get()` as the maps get accessed for other things?

So there may be some situations where it might help reduce garbage, although I'm not sure how much of a difference it makes in practice.  I think with each incoming event currently making its own `StorageScope`, the lifetime of the corresponding frame would be roughly the lifetime of the request, at which point the remaining key reference should get freed?  My understanding is that it would be hard for expired keys to end up, say, accumulating in the "root" frame's map.

But, yeah, I'm planning to move the tracing key to the isolate level, since it seems like it needs to be there in order to properly support non-durable-object workers (or, more generally, the case an isolate being shared between multiple DO/non-DO workers in disparate pipelines).  In which case, storage key cleanup might be moot, since it would all go away when the isolate does.

Reply:
Yes that's exactly the idea and yeah, if the lifespan of the key is expected to match the lifespan of the frame, then it should be fine. I would probably still recommend resetting just to be on the safe side.